src/Sulu/Bundle/ContentBundle/Preview/PreviewMessageHandler.php

```diff
@@ -0,0 +1,302 @@
+        } catch (\Exception $e) {
+            // send fail message
+            $conn->send(
+                json_encode(
+                    array(
+                        'command' => 'fail',
+                        'code' => $e->getCode(),
+                        'msg' => $e->getMessage(),
+                        'parentMsg' => $message
+                    )
+                )
+            );
+        }
+    }
+
+    /**
+     * Executes command
+     * @param ConnectionInterface $conn
+     * @param MessageHandlerContext $context
+     * @param array $msg
+     * @return mixed|null
+     * @throws ContextParametersNotFoundException
+     * @throws MissingParameterException
+     */
+    private function execute(ConnectionInterface $conn, MessageHandlerContext $context, $msg)
+    {
+        if (!array_key_exists('command', $msg)) {
+            throw new MissingParameterException('command');
+        }
+        $command = $msg['command'];
+        $result = null;
+
+        switch ($command) {
+            case 'start':
+                $result = $this->start($conn, $context, $msg);
+                break;
+            case 'stop':
+                $result = $this->stop($conn, $context);
+                break;
+            case 'update':
+                $result = $this->update($conn, $context, $msg);
+                break;
+        }
+
+        return $result;
+    }
+
+    /**
+     * Reconnect to mysql
+     */
+    private function reconnect()
+    {
+        /** @var EntityManagerInterface $entityManager */
+        $entityManager = $this->registry->getManager();
+        /** @var Connection $connection */
+        $connection = $entityManager->getConnection();
+
+        try {
+            $connection->executeQuery('SELECT 1;');
+        } catch (DBALException $exc) {
+            $this->logger->warning('Mysql reconnect');
+            $connection->close();
+            $connection->connect();
+        }
+    }
+
+    /**
+     * Start preview session
+     * @param ConnectionInterface $conn
+     * @param MessageHandlerContext $context
+     * @param array $msg
+     * @return array
+     * @throws MissingParameterException
+     */
+    private function start(ConnectionInterface $conn, MessageHandlerContext $context, $msg)
+    {
+        // init session
+
+        // locale
+        if (!array_key_exists('locale', $msg)) {
+            throw new MissingParameterException('locale');
+        }
+        $locale = $msg['locale'];
+        $context->set('locale', $locale);
+
+        // webspace key
+        if (!array_key_exists('webspaceKey', $msg)) {
+            throw new MissingParameterException('webspaceKey');
+        }
+        $webspaceKey = $msg['webspaceKey'];
+        $context->set('webspaceKey', $webspaceKey);
+
+        // user id
+        if (!array_key_exists('user', $msg)) {
+            throw new MissingParameterException('user');
+        }
+        $user = $msg['user'];
```

Comment:
The problem is that the websocket cannot use the filesystem session. therefore you have to send the user here ... when we use Memcached or ODB Sessions it would work with the `AuthentiatedMessageDispatcherApp` and the `$content->getAdminUser()` ... but for now it is enough to inject it here.